### PR TITLE
Feat: Attempt generate from multiple files

### PR DESCRIPTION
**Pull request by Alex-A4**
The main:
1) Added `bin/merger.dart`, that is used to merge `*api.rs` files.
2) Added doc with description how to form imports to let merger work correctly
3) A bit changed structure of logger and caller, so they inside `utils/` because it looks more clearly
4) Updated all generated files with `cargo build`

### CODE REVIEW

File: packages/nekoton_bridge/bin/merger.dart [hunk 1]
```diff
@@ -0,0 +1,200 @@
...
+              crates,
+            ),
+          );
+
+      /// Only code that should be saved to result file
+      final pureContent = content
+          // all imports
+          .replaceAll(importsReg, '')
+          // all mods
+          .replaceAll(RegExp('.*mod.*;'), '')
```
Comment (nesquikm) [inline packages/nekoton_bridge/bin/merger.dart:37]: Maybe do it consistently by taking out strings to constants, like mergedFileName?

File: packages/nekoton_bridge/bin/merger.dart [hunk 2]
```diff
@@ -0,0 +1,200 @@
...
+          // all allows
+          .replaceAll(RegExp(r'\#\!\[allow\(.*\)\]'), '')
```
Comment (nesquikm) [inline packages/nekoton_bridge/bin/merger.dart:39]: One more constant maybe?

File: packages/nekoton_bridge/native/.gitignore [hunk 3]
```diff
@@ -1,3 +1,7 @@
 # Rust library related
 Cargo.lock
 target
+
+# TODO: Uncomment when CI will be configured to ingore missed imports for rust
+#bridge_generated.rs
+#merged.rs
```
Comment (nesquikm) [inline packages/nekoton_bridge/native/.gitignore:7]: EOF newline ¯\_(ツ)_/¯

File: packages/nekoton_bridge/native/src/nekoton_wrapper/crypto/mnemonic/mnemonic_api.rs [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+
+use crate::nekoton_wrapper::{
+    crypto::mnemonic::models::KeypairHelper, models_api::GeneratedKeyG, str_list_to_string_vec,
+    str_vec_to_string_vec, HandleError, MatchResult,
+};
+
+pub use nekoton::crypto::MnemonicType;
+
+/// Generate seed phrase by specified mnemonic type
+pub fn nt_generate_key(account_type: MnemonicType) -> SyncReturn<GeneratedKeyG> {
```
Comment (nesquikm) [inline packages/nekoton_bridge/native/src/nekoton_wrapper/crypto/mnemonic/mnemonic_api.rs:12]: Why sync?

**Review by nesquikm — CHANGES_REQUESTED** [review]

File: packages/nekoton_bridge/native/build.rs [hunk 5]
```diff
@@ -33,7 +33,7 @@ fn main() {
 
     // Format the generated Dart code
     _ = std::process::Command::new("flutter")
-        .arg("format")
+        .arg("format -l 100")
```
[the remark this answers is not in the document]
Comment (Alex-A4) [reply packages/nekoton_bridge/native/build.rs:36]: -

File: packages/nekoton_bridge/native/src/utils/logger.rs [hunk 6]
```diff
@@ -175,17 +189,17 @@ impl Log for MyMobileLogger {
...
             record.to_builder().level(override_level).build()
         };
 
         #[cfg(target_os = "android")]
-        android_logger::log(&modified_record);
+            android_logger::log(&modified_record);
 
         #[cfg(target_os = "ios")]
-        self.ios_logger.log(&modified_record);
+            self.ios_logger.log(&modified_record);
```
[the remark this answers is not in the document]
Comment (Alex-A4) [reply packages/nekoton_bridge/native/src/utils/logger.rs]: -

File: packages/nekoton_bridge/native/src/utils/logger.rs [hunk 7]
```diff
@@ -205,4 +219,4 @@ impl SharedLogger for MyMobileLogger {
     fn as_log(self: Box<Self>) -> Box<dyn Log> {
         Box::new(*self)
     }
-}
+}
```
[the remark this answers is not in the document]
Comment (Alex-A4) [reply packages/nekoton_bridge/native/src/utils/logger.rs]: -

File: packages/nekoton_bridge/native/.gitignore [hunk 8]
```diff
@@ -1,3 +1,5 @@
 # Rust library related
 Cargo.lock
 target
+bridge_generated.rs
+merged.rs
```
[the remark this answers is not in the document]
Comment (Alex-A4) [reply packages/nekoton_bridge/native/.gitignore]: -

File: packages/nekoton_bridge/native/build.rs [hunk 9]
```diff
@@ -1,18 +1,18 @@
...
-const RUST_INPUT: &str = "src/api.rs";
+const RUST_INPUT: &str = "src/merged.rs";
 const DART_OUTPUT: &str = "../lib/src/bridge_generated.dart";
-
 const IOS_C_OUTPUT: &str = "../../flutter_nekoton_bridge/ios/Classes/frb.h";
 // const MACOS_C_OUTPUT: &str = "../../flutter_nekoton_bridge/macos/Classes/frb.h";
 
 fn main() {
-    // Tell Cargo that if the input Rust code changes, rerun this build script
-    println!("cargo:rerun-if-changed={RUST_INPUT}");
```
[the remark this answers is not in the document]
Comment (Alex-A4) [reply packages/nekoton_bridge/native/build.rs:13]: -

File: packages/nekoton_bridge/native/src/utils/mod.rs [hunk 10]
```diff
@@ -0,0 +1,3 @@
+pub mod api;
+pub mod caller;
+pub mod logger;
```
[the remark this answers is not in the document]
Comment (Alex-A4) [reply packages/nekoton_bridge/native/src/utils/mod.rs]: -

File: packages/nekoton_bridge/native/src/nekoton_wrapper/crypto/mod.rs [hunk 11]
```diff
@@ -0,0 +1,4 @@
+pub mod derived_key;
+pub mod encrypted_key;
+pub mod ledger_key;
+pub mod mnemonic;
```
[the remark this answers is not in the document]
Comment (Alex-A4) [reply packages/nekoton_bridge/native/src/nekoton_wrapper/crypto/mod.rs]: -

File: packages/nekoton_bridge/native/src/nekoton_wrapper/models_api.rs [hunk 12]
```diff
@@ -0,0 +1,19 @@
...
+
+/// -------------------------
+/// Types to generate in dart
+/// -------------------------
+
+/// Mirror struct of MnemonicType
+#[frb(mirror(MnemonicType))]
+pub enum _MnemonicType {
+    Legacy,
+    Labs(u16),
```
[the remark this answers is not in the document]
Comment (Alex-A4) [reply packages/nekoton_bridge/native/src/nekoton_wrapper/models_api.rs:12]: Shall we duplicate original nekoton doc?

File: packages/nekoton_bridge/native/src/nekoton_wrapper/crypto/encrypted_key/mod.rs [hunk 13]
```diff
@@ -0,0 +1,2 @@
+pub mod encrypted_key_api;
+pub mod models;
```
[the remark this answers is not in the document]
Comment (Alex-A4) [reply packages/nekoton_bridge/native/src/nekoton_wrapper/crypto/encrypted_key/mod.rs]: -

File: packages/nekoton_bridge/bin/merger.dart (hunk 2 above)
Comment (Alex-A4) [reply packages/nekoton_bridge/bin/merger.dart:39]: Do you think we should put it as global string?
It's used only in one place

File: packages/nekoton_bridge/native/.gitignore (hunk 3 above)
Comment (Alex-A4) [reply packages/nekoton_bridge/native/.gitignore:7]: -

File: packages/nekoton_bridge/native/src/nekoton_wrapper/crypto/mnemonic/mnemonic_api.rs [hunk 14]
```diff
@@ -0,0 +1,37 @@
...
+pub fn nt_get_hints(input: String) -> SyncReturn<Vec<String>> {
+    SyncReturn(str_list_to_string_vec(dict::get_hints(input.as_str())))
+}
+
+/// Generate public and secret keys from seed phrase and mnemonic type
+/// Returns json {'public': '...', 'secret': '...'}
+pub fn nt_derive_from_phrase(phrase: String, mnemonic_type: MnemonicType) -> SyncReturn<String> {
+    // Result<Keypair, Error> {
+    let keypair = derive_from_phrase(phrase.as_str(), mnemonic_type).handle_error();
+    SyncReturn(
```
[the remark this answers is not in the document]
Comment (Alex-A4) [reply packages/nekoton_bridge/native/src/nekoton_wrapper/crypto/mnemonic/mnemonic_api.rs:32]: -

File: packages/nekoton_bridge/native/src/nekoton_wrapper/crypto/mnemonic/mnemonic_api.rs [hunk 15]
```diff
@@ -0,0 +1,37 @@
...
+/// input: acco
+/// returns [account, accommodate, ...]
+pub fn nt_get_hints(input: String) -> SyncReturn<Vec<String>> {
+    SyncReturn(str_list_to_string_vec(dict::get_hints(input.as_str())))
+}
+
+/// Generate public and secret keys from seed phrase and mnemonic type
+/// Returns json {'public': '...', 'secret': '...'}
+pub fn nt_derive_from_phrase(phrase: String, mnemonic_type: MnemonicType) -> SyncReturn<String> {
+    // Result<Keypair, Error> {
```
[the remark this answers is not in the document]
Comment (Alex-A4) [reply packages/nekoton_bridge/native/src/nekoton_wrapper/crypto/mnemonic/mnemonic_api.rs:30]: -

File: packages/nekoton_bridge/native/src/nekoton_wrapper/crypto/mnemonic/mnemonic_api.rs (hunk 4 above)
Comment (Alex-A4) [reply packages/nekoton_bridge/native/src/nekoton_wrapper/crypto/mnemonic/mnemonic_api.rs:12]: -

File: packages/nekoton_bridge/native/src/nekoton_wrapper/crypto/mnemonic/mnemonic_api.rs [hunk 16]
```diff
@@ -0,0 +1,37 @@
...
+    SyncReturn(GeneratedKeyG {
+        words: str_vec_to_string_vec(key.words),
+        account_type: key.account_type,
+    })
+}
+
+/// Get hints for input part of word of seed phrase to get possible words
+/// input: acco
+/// returns [account, accommodate, ...]
+pub fn nt_get_hints(input: String) -> SyncReturn<Vec<String>> {
```
[the remark this answers is not in the document]
Comment (Alex-A4) [reply packages/nekoton_bridge/native/src/nekoton_wrapper/crypto/mnemonic/mnemonic_api.rs:23]: -

File: packages/nekoton_bridge/native/src/nekoton_wrapper/crypto/encrypted_key/encrypted_key_api.rs [hunk 17]
```diff
@@ -0,0 +1,9 @@
+pub use nekoton::crypto::MnemonicType;
+
+pub const ENCRYPTED_KEY_SIGNER_NAME: &str = "EncryptedKeySigner";
+
+
+/// Test func
+pub fn resend_mnemonic(mnemonic: MnemonicType) -> MnemonicType {
+    mnemonic
+}
```
[the remark this answers is not in the document]
Comment (Alex-A4) [reply packages/nekoton_bridge/native/src/nekoton_wrapper/crypto/encrypted_key/encrypted_key_api.rs]: -

File: packages/nekoton_bridge/native/src/nekoton_wrapper/models_api.rs (hunk 12 above)
[the remark this answers is not in the document]
Comment (nesquikm) [reply packages/nekoton_bridge/native/src/nekoton_wrapper/models_api.rs:12]: Nope, agreed

**Review by nesquikm — CHANGES_REQUESTED** [review]
Also you can use #[allow(dead_code)] or #![allow(dead_code)] to fix "unused code" warnings

File: packages/nekoton_bridge/bin/merger.dart (hunk 2 above)
Comment (nesquikm) [reply packages/nekoton_bridge/bin/merger.dart:48]: Just for consistency with other regex in the file maybe?

**Review by nesquikm — CHANGES_REQUESTED** [review]

**Review by nesquikm — APPROVED** [review]
It was tremendous piece of work, comraden!

**Pull request merged**